### ISSUE TITLE
7$: Improve error if string literal has closing smart quote

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#631]
In the following example, the string literal ends with a smart quote, leading to confusing errors:
```js
// hello.js:1:13: error: unclosed string literal [E0040]
// hello.js:1:24: error: expected ')' to close function call [E0141]
// hello.js:1:12: note: function call started here [E0141]
console.log("hello”);
```
When E0040 would be reported, recover by detecting the smart quote as the end, reporting an error like 'cannot close string with ”; use " instead' instead.

**Comment by strager** [comment]
> should we do similar thing for this code as well?

Yeah, that's a good idea.
